Python/initconfig.c

```diff
@@ -1813,7 +1838,7 @@ config_init_int_max_str_digits(PyConfig *config)
 #undef _STRINGIFY
 #undef STRINGIFY
         }
-        _Py_global_config_int_max_str_digits = maxdigits;
+        config->int_max_str_digits = maxdigits;
     }
```

Comment:
I suggest to compute the default in PyConfig, rather than in _PyLong_InitTypes(), and add a check in config_check_consistency() that once PyConfig is initialized, it's ``>= 0``.

```
    if (config->int_max_str_digits < 0) {
        config->int_max_str_digits = _PY_LONG_DEFAULT_MAX_STR_DIGITS;
    }
```